### ISSUE TITLE
IllegalStateException and eventual crash on start-up

### ISSUE THREAD

**Issue by SlimeDog** [Zrips/Residence#830]
Spigot 1.18.1-3378
CMILib 1.1.0.4
Residence 5.0.1.0

Things have been fine since the snakeyaml changes on Spigot, but tonight on server start-up
```
[22:53:51] [Server thread/INFO]: [Residence] Enabling Residence v5.0.1.0
[22:53:52] [Server thread/INFO]: [Residence] Disabling Residence v5.0.1.0
[22:53:52] [Server thread/INFO]: ^[[0;32;1m[^[[0;33;22mResidence^[[0;32;1m]^[[0;37;22m Loaded (2) groups^[[m
[22:53:52] [Server thread/INFO]: ^[[0;32;1m[^[[0;33;22mResidence^[[0;32;1m]^[[0;37;22m Found Vault using permissions plugin:LuckPerms^[[m
[22:53:52] [Server thread/INFO]: ^[[0;32;1m[^[[0;33;22mResidence^[[0;32;1m]^[[0;37;22m Found WorldEdit 7.2.9-SNAPSHOT+6034-92dfe5e^[[m
[22:53:52] [Server thread/INFO]: ^[[0;32;1m[^[[0;33;22mResidence^[[0;32;1m]^[[0;37;22m - FAILED INITIALIZATION! DISABLED! ERROR:^[[m
[22:53:52] [Server thread/ERROR]: null
java.lang.IllegalStateException: Queue full
        at java.util.AbstractQueue.add(AbstractQueue.java:98) ~[?:?]
        at java.util.concurrent.ArrayBlockingQueue.add(ArrayBlockingQueue.java:329) ~[?:?]
        at org.yaml.snakeyaml.emitter.Emitter.emit(Emitter.java:252) ~[snakeyaml-1.30.jar:?]
        at org.yaml.snakeyaml.serializer.Serializer.serializeComments(Serializer.java:224) ~[snakeyaml-1.30.jar:?]
        at org.yaml.snakeyaml.serializer.Serializer.serializeNode(Serializer.java:168) ~[snakeyaml-1.30.jar:?]
        at org.yaml.snakeyaml.serializer.Serializer.serializeNode(Serializer.java:206) ~[snakeyaml-1.30.jar:?]
        at org.yaml.snakeyaml.serializer.Serializer.serializeNode(Serializer.java:207) ~[snakeyaml-1.30.jar:?]
        at org.yaml.snakeyaml.serializer.Serializer.serializeNode(Serializer.java:207) ~[snakeyaml-1.30.jar:?]
        at org.yaml.snakeyaml.serializer.Serializer.serialize(Serializer.java:115) ~[snakeyaml-1.30.jar:?]
        at org.yaml.snakeyaml.Yaml.serialize(Yaml.java:366) ~[snakeyaml-1.30.jar:?]
        at org.bukkit.configuration.file.YamlConfiguration.saveToString(YamlConfiguration.java:88) ~[spigot-api-1.18.1-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.configuration.file.FileConfiguration.save(FileConfiguration.java:65) ~[spigot-api-1.18.1-R0.1-SNAPSHOT.jar:?]
        at com.bekvon.bukkit.residence.ConfigManager.UpdateFlagFile(ConfigManager.java:471) ~[?:?]
        at com.bekvon.bukkit.residence.Residence.onEnable(Residence.java:485) ~[?:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264) ~[spigot-api-1.18.1-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:342) ~[spigot-api-1.18.1-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:480) ~[spigot-api-1.18.1-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.craftbukkit.v1_18_R1.CraftServer.enablePlugin(CraftServer.java:526) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3378-Spigot-8965a50-0ba6b90]
        at org.bukkit.craftbukkit.v1_18_R1.CraftServer.enablePlugins(CraftServer.java:440) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3378-Spigot-8965a50-0ba6b90]
        at net.minecraft.server.MinecraftServer.loadWorld0(MinecraftServer.java:612) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3378-Spigot-8965a50-0ba6b90]
        at net.minecraft.server.MinecraftServer.loadLevel(MinecraftServer.java:414) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3378-Spigot-8965a50-0ba6b90]
        at net.minecraft.server.dedicated.DedicatedServer.e(DedicatedServer.java:262) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3378-Spigot-8965a50-0ba6b90]
        at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:994) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3378-Spigot-8965a50-0ba6b90]
        at net.minecraft.server.MinecraftServer.lambda$0(MinecraftServer.java:304) ~[spigot-1.18.1-R0.1-SNAPSHOT.jar:3378-Spigot-8965a50-0ba6b90]
        at java.lang.Thread.run(Thread.java:833) [?:?]
[22:53:52] [Server thread/ERROR]: Error occurred while enabling Residence v5.0.1.0 (Is it up to date?)
java.lang.IllegalStateException: zip file closed
        at java.util.zip.ZipFile.ensureOpen(ZipFile.java:831) ~[?:?]
        at java.util.zip.ZipFile.getEntry(ZipFile.java:330) ~[?:?]
        at java.util.jar.JarFile.getEntry(JarFile.java:518) ~[?:?]
        at java.util.jar.JarFile.getJarEntry(JarFile.java:473) ~[?:?]
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:159) ~[spigot-api-1.18.1-R0.1-SNAPSHOT.jar:?]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:587) ~[?:?]
        at org.bukkit.plugin.java.PluginClassLoader.loadClass0(PluginClassLoader.java:104) ~[spigot-api-1.18.1-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.java.PluginClassLoader.loadClass(PluginClassLoader.java:99) ~[spigot-api-1.18.1-R0.1-SNAPSHOT.jar:?]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
        at com.bekvon.bukkit.residence.Residence.getShopSignUtilManager(Residence.java:991) ~[?:?]
```
Server starts up without error if Residence is removed. On my servers, it takes 12 restarts to replicate the error. At which point the server crashes.

`flags.yml` was completely rewritten with multiple copies of the same comment lines. It looks like every restart adds another set. If I delete all the comments from `config.yml` and `flags.yml`, Residence loads without error.

Comment at 548
```
    # click2, click1, bow_fire, door_toggle, iron_door_toggle, trapdoor_toggle, iron_trapdoor_toggle, fence_gate_toggle, door_close, iron_door_close, trapdoor_close, iron_trapdoor_close, fence_gate_close, extinguish, record_play, ghast_shriek, ghast_shoot, blaze_shoot, zombie_chew_wooden_door, zombie_chew_iron_door, zombie_destroy_door, smoke, step_sound, potion_break, instant_potion_break, ender_signal, mobspawner_flames, brewing_stand_brew, chorus_flower_grow, chorus_flower_death, portal_travel, endereye_launch, firework_shoot, villager_plant_grow, dragon_breath, anvil_break, anvil_use, anvil_land, enderdragon_shoot, wither_break_block, wither_shoot, zombie_infect, zombie_converted_villager, bat_takeoff, end_gateway_spawn, enderdragon_growl, phantom_bite, zombie_converted_to_drowned, husk_converted_to_zombie, grindstone_use, book_page_turn, smithing_table_use, pointed_dripstone_land, pointed_dripstone_drip_lava_into_cauldron, pointed_dripstone_drip_water_into_cauldron, skeleton_converted_to_stray, composter_fill_attempt, lava_interact, redstone_torch_burnout, end_portal_frame_fill, dripping_dripstone, bone_meal_use, ender_dragon_destroy_block, sponge_dry, electric_spark, copper_wax_on, copper_wax_off, oxidised_copper_scrape, wither_spawned, ender_dragon_death, end_portal_created_in_overworld, phantom_bites, zombie_converts_to_drowned, husk_converts_to_zombie, grindstone_used, book_page_turned, composter_composts, lava_converts_block, redstone_torch_burns_out, ender_eye_placed, ender_dragon_destroys_block, wet_sponge_vaporizes_in_nether, 
```
is probably a culprit, as well.

**Comment by SlimeDog** [comment]
flags.yml was completely rewritten with multiple copies of the same comment lines.

**Comment by SlimeDog** [comment]
Comment at 548
```
    # click2, click1, bow_fire, door_toggle, iron_door_toggle, trapdoor_toggle, iron_trapdoor_toggle, fence_gate_toggle, door_close, iron_door_close, trapdoor_close, iron_trapdoor_close, fence_gate_close, extinguish, record_play, ghast_shriek, ghast_shoot, blaze_shoot, zombie_chew_wooden_door, zombie_chew_iron_door, zombie_destroy_door, smoke, step_sound, potion_break, instant_potion_break, ender_signal, mobspawner_flames, brewing_stand_brew, chorus_flower_grow, chorus_flower_death, portal_travel, endereye_launch, firework_shoot, villager_plant_grow, dragon_breath, anvil_break, anvil_use, anvil_land, enderdragon_shoot, wither_break_block, wither_shoot, zombie_infect, zombie_converted_villager, bat_takeoff, end_gateway_spawn, enderdragon_growl, phantom_bite, zombie_converted_to_drowned, husk_converted_to_zombie, grindstone_use, book_page_turn, smithing_table_use, pointed_dripstone_land, pointed_dripstone_drip_lava_into_cauldron, pointed_dripstone_drip_water_into_cauldron, skeleton_converted_to_stray, composter_fill_attempt, lava_interact, redstone_torch_burnout, end_portal_frame_fill, dripping_dripstone, bone_meal_use, ender_dragon_destroy_block, sponge_dry, electric_spark, copper_wax_on, copper_wax_off, oxidised_copper_scrape, wither_spawned, ender_dragon_death, end_portal_created_in_overworld, phantom_bites, zombie_converts_to_drowned, husk_converts_to_zombie, grindstone_used, book_page_turned, composter_composts, lava_converts_block, redstone_torch_burns_out, ender_eye_placed, ender_dragon_destroys_block, wet_sponge_vaporizes_in_nether, 
```
is probably a culprit.

**Comment by SlimeDog** [comment]
Seems to be resolved with CMILib 1.1.0.5. Thanks.
